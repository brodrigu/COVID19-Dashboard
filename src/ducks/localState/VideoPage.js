import { createAction, handleActions } from 'redux-actions';

// Action creators
export const setActiveTab = createAction('VideoPage/setActiveTab');
export const setDisplayedProduct = createAction('VideoPage/setDisplayedProduct');

// Reducers
export default handleActions({
    [setActiveTab](state, { payload }) {
        return { ...state, activeTab: payload };
    },
    [setDisplayedProduct](state, { payload }) {
        return { ...state, displayedProduct: payload };
    },
}, {
    activeTab: 1,
    displayedProduct: {},
});

// Selectors
export const getActiveTab = state => state.activeTab;
