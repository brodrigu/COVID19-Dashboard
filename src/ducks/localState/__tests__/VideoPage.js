/* eslint-env jest */
import configureStore from 'redux-mock-store';

// Actions to be tested
import * as VideoPageDuck from '../VideoPage';

const mockStore = configureStore();
const store = mockStore();

describe('VideoPage Actions', () => {
    beforeEach(() => {
        store.clearActions();
    });

    for (const action of [
        'setActiveTab',
        'setDisplayedProduct',
    ]) {
        test(`${action} test`, () => {
            store.dispatch(VideoPageDuck[action]());
            expect(store.getActions()).toMatchSnapshot();
        });
    }
});

describe('VideoPage Reducers', () => {
    beforeEach(() => {
    });

    test('initial state', () => {
        const action = { type: 'dummy_action' };
        expect(VideoPageDuck.default(undefined, action)).toMatchSnapshot();
    });

    test('setActiveTab', () => {
        const action = { payload: 8, type: `${VideoPageDuck.setActiveTab}` };
        expect(VideoPageDuck.default(undefined, action)).toMatchSnapshot();
    });

    test('setDisplayedProduct', () => {
        const action = { payload: 'myProduct', type: `${VideoPageDuck.setDisplayedProduct}` };
        expect(VideoPageDuck.default(undefined, action)).toMatchSnapshot();
    });
});
