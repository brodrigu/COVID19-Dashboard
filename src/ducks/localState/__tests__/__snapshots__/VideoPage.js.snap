// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPage Actions setActiveTab test 1`] = `
Array [
  Object {
    "type": "VideoPage/setActiveTab",
  },
]
`;

exports[`VideoPage Actions setDisplayedProduct test 1`] = `
Array [
  Object {
    "type": "VideoPage/setDisplayedProduct",
  },
]
`;

exports[`VideoPage Reducers initial state 1`] = `
Object {
  "activeTab": 1,
  "displayedProduct": Object {},
}
`;

exports[`VideoPage Reducers setActiveTab 1`] = `
Object {
  "activeTab": 8,
  "displayedProduct": Object {},
}
`;

exports[`VideoPage Reducers setDisplayedProduct 1`] = `
Object {
  "activeTab": 1,
  "displayedProduct": "myProduct",
}
`;
